test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,396 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package framework
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/onsi/ginkgo"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var seedRegistrationConfig *SeedRegistrationConfig
+
+const (
+	kubeconfigString = "kubeconfig"
+)
+
+// SeedRegistrationFramework represents the seed test framework that includes
+// test functions that can be executed on a specific seed
+type SeedRegistrationFramework struct {
+	*GardenerFramework
+	TestDescription
+	Config *SeedRegistrationConfig
+
+	SeedClient kubernetes.Interface
+	Seed       *gardencorev1beta1.Seed
+	Secret     *corev1.Secret //the secret that will be created and used for seed.Spec.SecretRef
+}
+
+// SeedRegistrationConfig is the configuration for a seed registration framework that will be filled with user provided data
+type SeedRegistrationConfig struct {
+	GardenerConfig         *GardenerConfig
+	SeedName               string
+	SecretBinding          string
+	BackupSecretProvider   string
+	ShootedSeedName        string
+	ShootedSeedNamespace   string
+	ShootedSeedKubeconfig  string
+	Provider               string
+	ProviderType           string
+	NetworkingDefServices  string
+	NetworkingDefPods      string
+	NetworkingPods         string
+	NetworkingServices     string
+	NetworkingkNodes       string
+	NetworkingkBlockCIDRs  string
```

Comment:
This field is currently only being validated but not used at all in buildSeedObject func.

Reply:
@kris94 Shouldn't this be used on seedregistrationframework.go:345
and if it is required
```suggestion
	NetworkingBlockCIDRs  string
```

Reply:
Yes, it should be